simpeg_drivers/utils/tile_estimate.py

```diff
@@ -0,0 +1,186 @@
+#  intend to preserve as trade secrets such software and information.
+#  This software is furnished pursuant to a written license agreement and
+#  may be used, copied, transmitted, and stored only in accordance with
+#  the terms of such license and with the inclusion of the above copyright
+#  notice.  This software and information or any other copies thereof may
+#  not be provided or otherwise made available to any other person.
+#
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+import sys
+from pathlib import Path
+from typing import ClassVar
+
+import matplotlib.pyplot as plt
+import numpy as np
+from geoapps_utils.driver.data import BaseData
+from geoapps_utils.utils.numerical import fibonacci_series, fit_circle
+from geoh5py.groups import SimPEGGroup, UIJsonGroup
+from geoh5py.ui_json import InputFile
+from geoh5py.ui_json.utils import fetch_active_workspace, monitored_directory_copy
+from scipy.interpolate import interp1d
+from tqdm import tqdm
+
+from simpeg_drivers import assets_path
+from simpeg_drivers.driver import DRIVER_MAP, InversionDriver
+from simpeg_drivers.utils.utils import (
+    active_from_xyz,
+    simpeg_group_to_driver,
+    tile_locations,
+)
+
+
+class TileEstimator:
+    def __init__(self, input_file: str | Path | InputFile):
+        if isinstance(input_file, str | Path):
+            input_file = InputFile.read_ui_json(input_file)
+
+        if not isinstance(input_file, InputFile):
+            raise ValueError(
+                "Input must be a path to a UI JSON file or an InputFile instance."
+            )
+
+        self.params = TileParameters.build(input_file)
+
+    def start(self) -> SimPEGGroup:
+        """
+        Run the tile estimator.
+        """
+        driver = simpeg_group_to_driver(self.params.simulation, self.params.geoh5)
+        mesh = driver.inversion_mesh.mesh
+        locations = driver.inversion_data.locations
+        active_cells = active_from_xyz(
+            driver.inversion_mesh.entity, locations, method="nearest"
+        )
+        results = {}
+        counts = fibonacci_series(15)[2:]
+        for count in tqdm(counts, desc="Estimating tiles:"):
+            tiles = tile_locations(
+                locations,
+                count,
+                method="kmeans",
+            )
+            # Get the median tile
+            ind = np.argsort([len(tile) for tile in tiles])[int(count / 2)]
+            survey, _, _ = driver.inversion_data.create_survey(
+                mesh=mesh, local_index=tiles[ind]
+            )
+            driver.params.tile_spatial = int(count)
+            local_sim, local_map = driver.inversion_data.simulation(
+                mesh,
+                active_cells,
+                survey,
+                padding_cells=driver.params.padding_cells,
+                tile_id=count,
+            )
+            results[count] = float(survey.nD) * local_map.shape[0] * count * 8 * 1e-9
```

Comment:
what do the `* 8` and `* 1e-9` represent?  Maybe assign them to a well named variable to make it clear?

Reply:
How to convert 32 bit float array to Gb

Reply:
oh, right, thanks.  Let's add that to a docstring in the refactor.